### PR TITLE
data: updates to the polymorphic relationships rfc

### PR DESCRIPTION
**Pull request by runspired**
[Rendered](https://github.com/emberjs/rfcs/blob/9de548960294e63d63f0b56e5a5fd2eeafea380a/text/0793-polymporphic-relations-without-inheritance.md)

While working on implementing polymorphic relationship support we discovered both that the restriction on polymorphic-to-polymorphic was unnecessary *and* that clarification of how to avoid needing to implement the abstract-type class was required. This updates the RFC text to reflect these things.

cc @yratanov @snewcomer

### CODE REVIEW

File: text/0793-polymporphic-relations-without-inheritance.md [hunk 1]
```diff
@@ -84,59 +84,149 @@ This RFC proposes to allow explicitly declaring polymorphic traits similar to [r
...
-In the below example, the polymporphic relationship implies the existence of the `base type` "taggable".
+In the below example, the polymporphic relationship implies the existence of the `abstract type` "taggable".
 
+```ts
+class Tag extends Model {
+  @hasMany("taggable", { async: false, inverse: "tags", polymorphic: true }) taggables;
+}
 ```
+
+It is only required that your app implement a model (or if using instantiateRecord and the schema service implement handling for that type) if your API will return entities whose type is the abstract type. If the abstract type is never directly interacted with, it is not required to implement a Model or other associated support.
```
Comment (yratanov) [inline text/0793-polymporphic-relations-without-inheritance.md:132]: > If the abstract type is never directly interacted with

What does it mean?

Comment (runspired) [reply text/0793-polymporphic-relations-without-inheritance.md:132]: This is actually carry over from the original rfc (I cleaned it up to refer to abstract vs base as it was more clear)

what it means is that if Post implements Taggable but no attempt to fetch something with type taggable and no data received will have type taggable then it’s never necessary to have a Model named Taggable

Comment (yratanov) [reply text/0793-polymporphic-relations-without-inheritance.md:132]: super! Thank you for clarification!

**Comment by ef4** [comment]
This looks ready to merge. Core framework looked it over briefly but I think it's up to the data team to merge it.

**Pull request merged**